### PR TITLE
Added CodeQL and Bandit security checks as GitHub Actions

### PR DESCRIPTION
**Pull request by mstfbl**
This PR adds the [CodeQL ](https://securitylab.github.com/tools/codeql/)and [Bandit ](https://bandit.readthedocs.io/en/latest/)Security checks in the form of GitHub Action Workflows, where the two workflows run with each opened PR to the `master` branch.

For CodeQL, the task will upload CodeQL results under the "Security" GitHub repo tab and in the "Code scanning results" section.

For Bandit, the task will pass if no high-severity issues have been found in the repo (outside of the `./third-party` submodules directory).

### CODE REVIEW

**Comment by mstfbl** [comment]
I see that the `ci/circleci: circleci_consistency` build test is failing. Per the given instructions, I ran `python .circleci/regenerate.py`, however I'm only seeing these probably wrong changes occur:

In `.circleci/config.yml:
```yml
workflows:
  build:
    jobs:
      - circleci_consistency
      - binary_linux_wheel:
          name: binary_linux_wheel_py3.6
          python_version: &#39;3.6&#39;
      - binary_linux_wheel:
          name: binary_linux_wheel_py3.7
          python_version: &#39;3.7&#39;
      - binary_linux_wheel:
          name: binary_linux_wheel_py3.8
          python_version: &#39;3.8&#39;
```

Whereas before it was:

```yml
workflows:
  build:
    jobs:
      - circleci_consistency
      - binary_linux_wheel:
          name: binary_linux_wheel_py3.6
          python_version: '3.6'
      - binary_linux_wheel:
          name: binary_linux_wheel_py3.7
          python_version: '3.7'
      - binary_linux_wheel:
          name: binary_linux_wheel_py3.8
          python_version: '3.8'
```

What is the correct fix for this?

**Comment by mstfbl** [comment]
I also see that the conda builds are failing as it can't find the `defusedxml` pip package I've added, while the wheel builds are passing. I added the `defusedxml` pip package as a requirement to both `requirements.txt` and to the linux and windows conda `environment.yml` files, however it seems like TorchText's CircleCI Conda builds do not use this, and instead use the build environment YMLs in `pytorch/builder`. Is there a reason why this is so?

File: .circleci/regenerate.py [hunk 1]
```diff
@@ -184,7 +185,7 @@ def unittest_workflows(indentation=6):
     env = jinja2.Environment(
         loader=jinja2.FileSystemLoader(d),
         lstrip_blocks=True,
-        autoescape=False,
+        autoescape=select_autoescape(enabled_extensions=('html', 'xml')),
```
Comment (malfet) [inline .circleci/regenerate.py:188]: Not that there is anything bad with this change, but `.circleci` folder is not redistributed with the product

File: .circleci/utils/test_sort_yaml.py [hunk 2]
```diff
@@ -11,4 +11,4 @@
 import sys
 import yaml
 
-sys.stdout.write(yaml.dump(yaml.load(sys.stdin, Loader=yaml.FullLoader), sort_keys=True))
+sys.stdout.write(yaml.dump(yaml.safe_load(sys.stdin, Loader=yaml.FullLoader), sort_keys=True))
```
Comment (malfet) [inline .circleci/utils/test_sort_yaml.py:14]: Thank you for the fix, but again, this file is not part of `torchvision` package

**Review by malfet — APPROVED** [review]
Looks good to me

**Comment by mstfbl** [comment]
I have removed the Bandit-specific changes from this PR. I've opened #1277 for the autoescape fix, #1278 for the `yaml.safe_load` fix, and #1279 for the `defusedxml` addition.

**Review by seemethere — APPROVED** [review]

**Pull request merged**